### DISCUSSION TITLE
Consistent use of dashboards

### DISCUSSION THREAD

**Discussion by pdwonline** [home-assistant/frontend#11778] [Other feature requests]
Recently, the energy monitoring feature of HA became configurable as a dashboard. Excellent choice.
A dashboard can be considered as a canvas specific for multiple items. If I look at this, HA would be much more consistent and flexible if we apply some basisc rules to what a dashboard would be and what it can do:

1. Dashboards have a title and an Icon that appeas in the left menu. This defines consistenly what the left menu items are (dashboards)
*  Make ALL left menu items a dashboard, including Map, history, logbook, media
*  Make dashboards editable (at least the title and icon) in the dashboards list
*  Make it possible to order the dashboards in the dashboards list; the left menu will be ordered accordingly

2. Dashboards can have tabs, just like the overview dashboard
* Users can change the name and icon of a tab
* Users can change the order of the tab
* Users can add a tab (just like the overview dashboard) to any dashboard.

This enables users to group functionality to a dashboard. For example an energy dashboard dedicated to energy, where one tab would show the standard energy dashboard as we currently know, and another tab could show individual energy meters and controls  as in the general view.

3. Dashboards can be hidden
* Per user. Just like you can define a tab in the overview dashboard to be visible for particular users or not. 
* Hidden dashboards do not show in the left menu. All dashboards will be shown in the dashboard list at all times. 
* System settings (could also be a dashboard, including the list of dashboards) can not be hidden for admins

This enables us to manage what is default shown through the left menu, indepenent to how the app will be configured.

**Comment by kongo09** [comment]
Along those lines:

Everything inside such a dashboard should be a card.

That would allow more streamlined configuration and extension, also of canned dashboards like the energy dashboard. As a user, it is difficult to understand why I configure an energy dashboard inside the main configuration settings, while I configure other dashboards in place.

**Comment by bedbug1226** [comment]
This needs to be a thing so bad.
